dashboard/test/ui/features/teacher_tools/teacher_dashboard/teacher_dashboard_progress_v2.feature

```diff
@@ -32,3 +32,37 @@ Scenario: Teacher can open and close Icon Key and details
   And I wait until element "h3:contains(Progress Tracking Icon Key)" is visible
   And I click selector "#ui-close-dialog"
   And element "h3:contains(Progress Tracking Icon Key)" is hidden
+
+@eyes
+Scenario: Teacher can view lesson progress for when students have completed a lesson and when they have started a lesson but not finished
+  Given I create an authorized teacher-associated student named "Sally"
+  Given I am assigned to unit "allthethings"
+  # Student completes one of many levels in lesson 2
+  And I complete the level on "http://studio.code.org/s/allthethings/lessons/2/levels/1"
+
+  # Student completes all the levels in lesson 10 (there is only one level)
+  Given I am on "http://studio.code.org/s/allthethings/lessons/10/levels/1?noautoplay=true"
+  Then I wait for 3 seconds
+  And I wait until element ".submitButton" is visible
+  And I press ".answerbutton[index=1]" using jQuery
+  And I press ".answerbutton[index=0]" using jQuery
+  And I press ".submitButton:first" using jQuery
+  And I wait to see ".modal"
+
+  When I sign in as "Teacher_Sally" and go home
+  And I get levelbuilder access
```

Comment:
Why does Sally need levelbuilder access to view the progress table in an eyes test?

Reply:
Yeah, this took me a while to understand as well... This was kinda a rabbit hole, but here is what I learned. For people with ONLY teacher permissions, the allthethings course is not assignable. However, if you have levelbuilder permissions, everything is assignable (see this [file](https://github.com/code-dot-org/code-dot-org/blob/6051a91aa58f48fd7d6e109fb1c6cc498d4903c6/dashboard/app/models/concerns/curriculum/assignable_course.rb#L4)). [ This file](https://github.com/code-dot-org/code-dot-org/blob/6051a91aa58f48fd7d6e109fb1c6cc498d4903c6/dashboard/config/courses/allthethingscourse.course) shows the data for `allthethings` - it is kind of a maze to trace through though.